apps/checker/app/matchers/DictionaryMatcher.scala

```diff
@@ -33,18 +33,35 @@ class DictionaryMatcher(
 
   val matcher = new LanguageToolMatcher(instance)
 
+  def matchFallsWithinNamedEntityRange(
+      ruleMatch: RuleMatch,
+      entities: List[EntityInText]
+  ): Boolean = {
+    entities.exists(entity =>
+      entity.range.from == ruleMatch.fromPos && entity.range.to == ruleMatch.toPos
+    )
+  }
+
   override def check(
       request: MatcherRequest
-  )(implicit ec: ExecutionContext) = {
-    // groupKey is used to control how rules are grouped in the client when they produces matches.
-    // This is needed for dictionary matches as they all share a common rule ID (MORFOLOGIK_RULE_COLLINS)
-    // groupKeys for dictionary matches have the format `MORFOLOGIK_RULE_COLLINS-{matchedText}`
+  )(implicit ec: ExecutionContext): Future[List[RuleMatch]] = {
+    val entityHelper = new EntityHelper()
+    val namedEntities =
+      request.blocks.flatMap((block) => entityHelper.getEntitiesFromText(block.text))
```

Comment:
Worth noting that this code will run the entity finder first, and then the check.

Might be nice to do that work in parallel. We can wrap blocking code in a `Future` to run it in another thread:
```suggestion
      Future { request.blocks.flatMap((block) => entityHelper.getEntitiesFromText(block.text)) }
```

We can then run both futures and perhaps use a for-comprehension to wait for them both before continuing:

```scala
for {
  result1 <- future1
  result2 <- future2
} yield {
  // do something w/ result1, result2
}
```

If we migrate to making HTTP calls to an external service, we'll likely get that for free, so perhaps we don't need to optimise yet.

Reply:
Great suggestion, I've made the change: https://github.com/guardian/typerighter/pull/443/commits/19244d3ff8de44f45c22413886a7609026241445